python/src/scipp/compat/mantid.py

```diff
@@ -465,28 +461,22 @@ def convert_Workspace2D_to_data_array(ws, **ignored):
     coords_labs_data = _convert_MatrixWorkspace_info(ws)
     _, data_unit = validate_and_get_unit(ws.YUnit(), allow_empty=True)
     coords_labs_data["data"] = sc.Variable([spec_dim, dim],
-                                           shape=(ws.getNumberHistograms(),
-                                                  len(ws.readY(0))),
                                            unit=data_unit,
-                                           variances=True)
+                                           values=ws.extractY(),
+                                           variances=ws.extractE())
     array = detail.move_to_data_array(**coords_labs_data)
 
     if ws.hasAnyMaskedBins():
         array.masks["bin"] = detail.move(
             make_bin_masks(common_bins, spec_dim, dim, ws.blocksize(),
                            ws.getNumberHistograms()))
         bin_masks = array.masks["bin"]
-        # set all the bin masks now - they're all the same
         if common_bins:
             set_common_bins_masks(bin_masks, dim, ws.maskedBinsIndices(0))
+        else:
+            for i in range(ws.getNumberHistograms()):
```

Comment:
This is not exactly equivalent to what is below, as we were checking `if ws.hasMaskedBins(i)` for every `i` index, while here we are only checking `if ws.hasAnyMaskedBins()`
Is this still ok?

Reply:
I was about to write "Yes it should, because the list is just empty in that case." But double checking indeed reveals that Mantid does *not* return an empty vector, but throws...